File: solutions/dart/armstrong-numbers/1/lib/armstrong_numbers.dart
class ArmstrongNumbers {
  bool isArmstrongNumber(String number){
    var lista = number.split("").map((e) => int.parse(e));
    int length = lista.length;
    int increment = 0;
    for (var element in lista)  {
      increment += (element * length);
    }
    return (increment == int.parse(number));
  }
}

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
import java.util.Arrays;
import java.util.List;

class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        double total = Arrays
                .stream(String.valueOf(numberToCheck).split(""))
                .map(Integer::valueOf)
                .toList()
                .stream()
                .map(x -> Math.pow(x, String.valueOf(numberToCheck).length()))
                .reduce(0d, Double::sum);
        return total == numberToCheck;
    }

}
